### DISCUSSION TITLE
Confirmation modal called by multiple instances of a Livewire component

### DISCUSSION THREAD

**Discussion by jgautier2** [wire-elements/modal#449] [Q&A]
Hi,

I use a Livewire component (called AddItemsTable) to build and show several tables on a page of my app.
So I have multiple instances of this class on the same page.
On each row of these different instances / tables, I have a delete button that uses a confirmation modal : the delete method of my AddItemsTable use $this->askForConfirmation(...) and a callback to process the deleting action.
My problem : when I confirm the confirmation modal , ConfirmationModal class dispatches 'actionConfirmed' to AddItemsTable class so every instance of it receive and react to it, so I have bugs.
How would handle this configuration ?
Is there a way to target a specific instance of my AddItemsTable components in the page ?
Maybe I miss something, so please let me know if i made a mistake somewhere.
Thanks a lot for any help.
JG

**Comment by joffreypersia** [comment]
@PhiloNL please, have you an idea ? Thank you :)

**Comment by Lirux** [comment]
Did you add a "wire:key" to the element in to loop?

See here:
https://livewire.laravel.com/docs/troubleshooting#adding-wirekey
